docs/reference/contributing/guidelines/style.md

```diff
@@ -59,55 +59,57 @@ uint32_t adc_function(analogin_t *obj, uint32_t options)
 ```
 ##### Rules
 
-* Indentation - four spaces. Please do not use tabs.
+- Indentation - four spaces. Please do not use tabs.
 
-* Braces - K&R style.
+- Braces - K&R style.
 
-* One true brace style (1TBS) - use braces for statements of type `if`, `else`, `while` and `for` (exception [from K&R](http://en.wikipedia.org/wiki/Indent_style#Variant:_1TBS)).
+- One true brace style (1TBS) - use braces for statements of type `if`, `else`, `while` and `for` (exception [from K&R](http://en.wikipedia.org/wiki/Indent_style#Variant:_1TBS)).
 
-* One line per statement.
+- One line per statement.
 
-* Preprocessor macro starts at the beginning of a new line; the code inside is indented according to the code above it.
+- Preprocessor macro starts at the beginning of a new line; the code inside is indented according to the code above it.
 
-* Cases within `switch` are indented (exception from K&R).
+- Cases within `switch` are indented (exception from K&R).
 
-* Space after statements of type `if`, `while`, `for`, `switch`. The same applies to binary operators (like, `+` and `*`) and the ternary operator (`?` and `:`).
+- Space after statements of type `if`, `while`, `for`, `switch`. The same applies to binary operators (like, `+` and `*`) and the ternary operator (`?` and `:`).
 
-* Each line preferably has at most 120 characters.
+- Each line preferably has at most 120 characters.
 
-* Comments should use proper spelling and grammar.
+- Comments should use proper spelling and grammar.
 
-* For pointers, `*` is adjacent to a name (`analogin_t *obj`).
+- For pointers or references, the symbols `*` or `&` are adjacent to a name (`analogin_t *obj`. `analogin_t &obj`). If you omit the name, place the space between the type and `*` (such as `int *` or `int &`).
 
-* Don't leave trailing spaces at the end of lines.
+- For return pointers or references, the symbols `*` or `&` are adjacent to a return type (`int* func()` or `int& func()`).
```

Comment:
`function return` or similar ? @AnotherButler please review

Reply:
This seems inconsistent with both variable declarations and type expressions.

``` cpp
int* doit(int *);
              ^--- right side?
   ^-------------- left side?
```

``` cpp
(struct { int *a; })* doit(int *);
                               ^--- right side?
                    ^-------------- left side?
              ^------------------- right side?                  
```

``` cpp
int (* doit(int *))(int *);
     ^------------------- what do we do here?
```

Reply:
I guess my qualms are more about how declarations are inconsistent with this rule.

In C, functions declarations are the same as other declarations, except the name has an argument list appended.

It is strange that they would have different style rules:
``` cpp
int hi;   // variable declaration
int hi(); // function declaration

int *hi;   // variable declaration
int* hi(); // function declaration
```

Reply:
@geky  We had this discussion for critical section a while ago (2 or 3 years back). I only captured what is in the codebase, as Kevin highlighted the older code has it this way, the newer code has it differently.

Based on the feedback and closer look at the codebase (mainly in the file handles and Buss clases there are pointers and references, its very few of the older code base and not even consistent there), let's make it `right`. I'll rebase this PR